doc: mcuboot: Clean up terminology for the 'direct-xip' mode. (#8502)
nrfconnect/sdk-nrf: doc/mcuboot/readme-ncs.rst

  When this file is programmed to the device, MCUboot will trigger the DFU procedure upon reboot.

.. note::

>    When you use MCUboot in the direct-xip mode, enable :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` to let the build system generate an additional set of files for the second application slot.

Reviewer: ```suggestion
   When you use MCUboot in the direct-xip mode, enable the :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` Kconfig option to let the build system generate an additional set of files for the second application slot.
```

Author: Fixed

   These ``.hex`` files are identical to the ones listed above, but their names also use the ``mcuboot_secondary_`` prefix.
   For example, :file:`mcuboot_secondary_app_signed.hex` is created and placed in the second slot on the target device when the :file:`app_signed.hex` file is placed in the first slot.

>    For more information about the direct-xip mode, see :ref:`mcuboot:direct-xip`.

Reviewer: ```suggestion
   For more information about the direct-xip mode, See the *Equal slots (direct-xip)* section in the :doc:`MCUboot documentation <mcuboot:design>`.
```
Unfortunately, we cannot have a direct link to that section (because the upstream markdown doc is messed up). I think this should do, at least for now.

Author: Fixed as suggested

>    For more information about the direct-xip mode, see the *Equal slots (direct-xip)* section in the :doc:`MCUboot documentation <mcuboot:design>`.

Reviewer: ```suggestion
   For more information about the direct-xip mode, see the *Equal slots (direct-xip)* section in the :doc:`Bootloader documentation <design>`.
```
Now that both ends (link start and target) are both in MCUboot, I believe this could work. At least it is worth trying.

Author: done
